lib/node_modules/@stdlib/repl/lib/multiline_handler.js

```diff
@@ -83,6 +84,12 @@ function MultilineHandler( repl, ttyWrite ) {
 	// Cache a reference to the command queue:
 	this._queue = repl._queue;
 
+	// Initialize an internal object for command history:
+	this._history = {};
```

Comment:
Would it make sense to create a new `History` class, rather than overloading the `MultilineHandler`? I suppose that there would be a circular dependency based on the current implementation. My forward looking concern is that we keep adding behavior to the multi-line handler. It's possible that it's unavoidable to have a mega-class which does everything, but this could introduce maintainability concerns in the future.

Reply:
The reason I did this is there is quite a bit of overlap with the keypresses. In some cases, the up arrow means go up a line, in some cases it means bring the previous command. Plus history depends on utilizing the class's `pasteMode` as well along with other input buffers. Moving history into a separate class would make this logic harder to implement. In general, I think `history` and `bracketedPaste` logically depend on the multiline handler class and are better implemented together. Maybe we can rename the `multilineHandler` class to something generic as its mostly handling everything related to how input is processed?

Reply:
Understood. Probably fine to leave as is, for now. There are likely design patterns to address your concerns, but this would likely require a fairly significant refactoring.